includes/orders/functions/orders.php

```diff
@@ -294,7 +294,14 @@ function edd_destroy_order( $order_id = 0 ) {
 		}
 	}
 
-
+	/**
+	 * Action hook for developers to do extra work when an order is destroyed.
+	 *
+	 * @since 3.0
+	 * @param int  $order_id  The original order ID.
+	 * @param bool $destroyed Whether the order was destroyed.
+	 */
+	do_action( 'edd_order_destroyed', $order_id, $destroyed );
```

Comment:
@cklosowski I think what I would prefer to do over this is to add a more extendable hook within the BerlinDB query class `delete_item` method. This, for example:

```php
do_action( $this->apply_prefix( "{$this->item_name}_deleted" ), $item_id, $result );
```

Running `edd_delete_order` would then automatically generate the following hooks straight away:
- `edd_order_deleted`
- `edd_order_item_deleted`
- `edd_order_address_deleted`

Since that is in a BerlinDB class, though, I'm not confident it would be safe to do, as it doesn't exist as a hook in BerlinDB at this time. If you think it's worth doing, though, I can submit a PR there.

Reply:
I think it is an interesting idea and worth pursuing. Let's go ahead and open a PR on BerlinDB for it, but keep the code in EDD core as you have done it here for the time being.

We've already had discussions about getting up to the latest BerlinDB for EDD 3.1, and if we can work to get this into a future BerlinDB release, then we can take advantage of it.